### ISSUE TITLE
NC 13 - Browsing external storage SMB/CIFS from "Files" app does not occur with simple left click on said directory

### ISSUE THREAD

**Issue by octo777** [nextcloud/server#7649]
### Steps to reproduce
1.  Create an external storage using SMB/CIFS
2.Verify that the settings were accepted and connection is established via Green light
3. Browse to Home directory in "Files" app. 
4. Click on the folder for any of the SMB shares

### Expected behaviour
When clicking on the folder represeting the SMB share it should go into that directory

### Actual behaviour
Nothing happens on the click event. 
However, the link associated is correct and copying the link location, or right click open in new tab does browse to that location correctly.  In fact, I just right clicked, "Open Link" and it even opened it in the current window. But just a plain old left click no worky worky. 
OMG I hope I get a shirt  :)

### Server configuration

**Operating system**:
Arch Linux

**Web server:**
Server version: Apache/2.4.29 (Unix)

**Database:**
mysql  Ver 15.1 Distrib 10.1.29-MariaDB, for Linux (x86_64) using readline 5.1

**PHP version:**
PHP 7.2.0

**Nextcloud version:** (see Nextcloud admin page)
13.0.0 Beta 3 directly from Github on 12/29/17
**Updated from an older Nextcloud/ownCloud or fresh install:**
Copied into place, but using an existing NC 12 database, so I would call it an upgrade.
It went through the App upgrade process splash screen

**Where did you install Nextcloud from:**
13.0.0 Beta 3 directly from Github on 12/29/17

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
Results
=======
- core
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- dav
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- files_external
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- federatedfilesharing
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- workflowengine
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- sharebymail
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- systemtags
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- files_sharing
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- twofactor_backupcodes
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- files
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- user_ldap
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- oauth2
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- comments
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- federation
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- admin_audit
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- files_versions
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- theming
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- files_trashbin
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- lookup_server_connector
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- updatenotification
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- provisioning_api
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.
- encryption
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\InvalidSignatureException
		- Signature data not found.

Raw output
==========
Array
(
    [core] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [dav] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [files_external] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [federatedfilesharing] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [workflowengine] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [sharebymail] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [systemtags] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [files_sharing] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [twofactor_backupcodes] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [files] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [user_ldap] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [oauth2] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [comments] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [federation] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [admin_audit] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [files_versions] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [theming] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [files_trashbin] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [lookup_server_connector] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [updatenotification] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [provisioning_api] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

    [encryption] => Array
        (
            [EXCEPTION] => Array
                (
                    [class] => OC\IntegrityCheck\Exceptions\InvalidSignatureException
                    [message] => Signature data not found.
                )

        )

)

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
Enabled:
  - admin_audit: 1.3.0
  - bookmarks: 0.10.1
  - bruteforcesettings: 1.0.3
  - calendar: 1.5.7
  - comments: 1.3.0
  - contacts: 2.0.1
  - dav: 1.4.5
  - deck: 0.2.7
  - external: 3.0.0
  - federatedfilesharing: 1.3.1
  - federation: 1.3.0
  - files: 1.8.0
  - files_accesscontrol: 1.3.0
  - files_automatedtagging: 1.3.0
  - files_downloadactivity: 1.2.0
  - files_external: 1.4.1
  - files_markdown: 2.0.1
  - files_retention: 1.2.0
  - files_sharing: 1.5.0
  - files_trashbin: 1.3.0
  - files_versions: 1.6.0
  - lookup_server_connector: 1.1.0
  - notes: 2.3.2
  - oauth2: 1.1.0
  - ocsms: 1.12.2
  - passman: 2.1.4
  - provisioning_api: 1.3.0
  - richdocuments: 1.12.39
  - sharebymail: 1.3.0
  - spreed: 2.9.0
  - systemtags: 1.3.0
  - theming: 1.4.1
  - twofactor_backupcodes: 1.2.3
  - updatenotification: 1.3.0
  - workflowengine: 1.3.0
Disabled:
  - encryption
  - testing
  - user_ldap

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system


PHP Warning:  PHP Startup: mcrypt: Unable to initialize module
Module compiled with module API=20160303
PHP    compiled with module API=20170718
These options need to match
 in Unknown on line 0
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "nextcloud.octra.pw"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "https:\/\/nextcloud.octra.pw\/nextcloud\/",
        "dbtype": "mysql",
        "version": "13.0.0.8",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "logtimezone": "UTC",
        "installed": true,
        "appstore.experimental.enabled": true,
        "theme": "",
        "loglevel": 2,
        "maintenance": false,
        "updater.release.channel": "stable",
        "enable_previews": true,
        "preview_max_filesize_image": 50,
        "preview_max_scale_factor": 10
    }
}

```
**Are you using external storage, if yes which one:  Yes, a SMB share

**Are you using encryption:** No

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:**  Firefox, Vivaldi, Chromium

**Operating system:**
 Arch Linux

#### Web server error log
<details>
<summary>Web server error log</summary>

```
There is no entry created in the log file when this error occurs, I watched it.
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
{"reqId":"DaqniyvMxCdIKb6IKehu","level":3,"time":"2017-12-29T20:28:34+00:00","remoteAddr":"10.0.50.50","user":"octoadmin","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/settings\/apps","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"DPF14r3jIHwaX8O5ODvW","level":3,"time":"2017-12-29T20:28:34+00:00","remoteAddr":"10.0.50.50","user":"octoadmin","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/avatar\/octoadmin\/32?v=0","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"JgL2Aebq6fXaKJKz06vk","level":3,"time":"2017-12-29T20:28:35+00:00","remoteAddr":"10.0.50.50","user":"--","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/apps\/theming\/favicon\/settings?v=2","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"xHOl1uL0dsx2PrQEYqat","level":3,"time":"2017-12-29T20:28:36+00:00","remoteAddr":"10.0.50.50","user":"octoadmin","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/settings\/apps\/categories","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"ENUFmQyuG0Zh7Wuk5ae9","level":3,"time":"2017-12-29T20:28:38+00:00","remoteAddr":"10.0.50.50","user":"octoadmin","app":"PHP","method":"GET","url":"\/nextcloud\/index.php\/settings\/apps\/list?category=installed","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"4tY2eEO8KQRBODIQZQEz","level":3,"time":"2017-12-29T20:28:39+00:00","remoteAddr":"10.0.50.50","user":"octoadmin","app":"PHP","method":"PROPFIND","url":"\/nextcloud\/remote.php\/webdav\/","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/62.0.3202.97 Safari\/537.36 Vivaldi\/1.95.1047.3","version":"13.0.0.8"}
{"reqId":"GmWJ7bdWdnZytQ0SDR7A","level":3,"time":"2017-12-29T20:29:10+00:00","remoteAddr":"10.0.50.50","user":"octo","app":"PHP","method":"PROPFIND","url":"\/nextcloud\/remote.php\/dav\/files\/octo\/","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (Linux) mirall\/2.3.3 (Nextcloud)","version":"13.0.0.8"}
{"reqId":"cTBLxvBY862uPU1il9wi","level":3,"time":"2017-12-29T20:29:25+00:00","remoteAddr":"10.0.50.50","user":"--","app":"PHP","method":"GET","url":"\/nextcloud\/status.php","message":"PHP Startup: mcrypt: Unable to initialize module\nModule compiled with module API=20160303\nPHP    compiled with module API=20170718\nThese options need to match\n at Unknown#0","userAgent":"Mozilla\/5.0 (Linux) mirall\/2.3.3 (Nextcloud)","version":"13.0.0.8"}

**Comment by MorrisJobke** [comment]
Looks like this is fixed by #7658

**Comment by MorrisJobke** [comment]
> Looks like this is fixed by #7658

cc @danxuliu for checking if this is fixed by #7658

**Comment by danxuliu** [comment]
@MorrisJobke Yes, #7658 should fix clicking on an external storage no matter the backend.

**Comment by MorrisJobke** [comment]
Fixed by #7658
